### PR TITLE
feat: update ui components, data fetching, tooling, map

### PR DESCRIPTION
**Pull request by stefanprobst**
this pr:

- introduces vue-query for request deduplication and response caching (see https://tanstack.com/query/latest/docs/framework/vue/overview)
- generates a new api client based on openapi spec document (see https://openapi-ts.pages.dev/)
- replaces vuetify ui components with radix/shadcn (see https://www.shadcn-vue.com/)
- replaces leaflet with maplibre-gl for the map view (see https://maplibre.org/maplibre-gl-js/docs/)
- updates tooling
- adds team page

it does not (yet) change our ci/cd workflow.

TODOs before this can be merged:

- [x] entity details page needs layout
- [x] display popover when clicking a geojson feature on the map (with links to entity details page)
- [ ] review config options exposed via cms
- [x] update github environment variables/secrets
- [x] ensure loading spinner on map page is visible in both light and dark mode
- [ ] review readme
- [ ] decide if it is actually ok to fetch all map features in one go (with `?limit=0`)
- [ ] mobile styles
- [ ] review where "brand" color (cms-configurable) should apply
- [ ] figure out intention for `type` relations on entity details page

FUTURE:

- [ ] needs tests for data and map pages
- [ ] we could create two additional branches (something like `variant/single-locale` and `variant/light-mode-only`), which make it easy for forks to remove i18n and/or dark-mode functionality
- [ ] data table: add column sorting (with tanstack/vue-table), improve pagination styles, make mobile friendly
- [ ] create some component abstractions on top of maplibre-gl
- [ ] maybe come up with a way to use the api response for `/map` to avoid fetching data for entity details pages. simply adding the following [here](https://github.com/acdh-oeaw/openatlas-discovery/blob/feat/update-all-the-things/composables/use-get-search-results.ts#L124) will not work because `setQueryData` is synchronous, and will block the main thread for ~2.5s for ~3000 geojson features:
  ```ts
  entity.features.forEach((feature) => {
    const params = { entityId: feature.properties._id };
    queryClient.setQueryData(["entity", params], entity);
  });
  ```
  EDIT: no longer viable, since we now use the `show` search param to select which fields to include in responses (and we limit to `system_class=place` on the map view)

### CODE REVIEW

File: components/entity-descriptions-display.vue [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+</script>
+
+<template>
+	<template v-if="filteredDescriptions.length > 0">
+		<div
+			v-for="(description, index) in filteredDescriptions"
+			:key="description.value ?? 'descr' + index"
+		>
+			<template v-if="description.value">
+				<VDivider v-if="index > 0" />
```
Comment (stefanprobst) [inline components/entity-descriptions-display.vue:22]: @Mocca101 can we replace this vuetify component with a plain `<div role="separator">`?

Comment (Mocca101) [reply components/entity-descriptions-display.vue:22]: Oh yes, sure!

**Comment by Mocca101** [comment]
@stefanprobst regarding the system class translations: The reason the translations are sometimes mapped to the same string, e.g. multiple to "Document", is due to it having been implemented like that, I believe intentionally, in previous Projects. So I think it might be good to check what exactly the desired behaviour is, or have it be configurable, if you haven't already done so anyways.

**Comment by Mocca101** [comment]
@stefanprobst regarding the system class translations: The reason the translations were sometimes mapped to the same string, e.g. multiple to "Document", is due to it having been implemented like that, I believe intentionally, in previous Projects. So I think it might be good to check what exactly the desired behaviour is, or have it be configurable, if you haven't already done so anyways.

**Comment by stefanprobst** [comment]
> I think it might be good to check what exactly the desired behaviour is

for now, i am going by https://demo.openatlas.eu/overview/model/openatlas_class_index (which is the same as the response from `/api/classes`). also posted a question in the openatlas mattermost channel about this.

**Comment by Mocca101** [comment]
> > I think it might be good to check what exactly the desired behaviour is
> 
> for now, i am going by https://demo.openatlas.eu/overview/model/openatlas_class_index (which is the same as the response from `/api/classes`). also posted a question in the openatlas mattermost channel about this.

Ok perfect, thank you!

**Pull request merged**